src/raster_types.jl

```diff
@@ -12,7 +13,24 @@ function Raster(dataset::GDALDatasetH)
                     _getrastercount(dataset)) # number of bands
     finalizer(raster, closeraster)
     raster
-end 
+end
+
+function Base.show(io::IO,raster::Raster)
+     if (raster.dataset == C_NULL)
+         println(io, raster.dataset)
+         return null;
```

Comment:
```
type Raster # simplified type for self-contained e.g.
    dataset::Int
    width::Int
    height::Int
    nband::Int
end

_bandtype(x, y) = typeof((x, y)) # simplified method for self-contained e.g.
````
This part should be included too? Or its just in your branch? Is there a way I can make the changes in the same PR or should I close and make a new one?